### PR TITLE
fix: correct row count for object store chunks

### PR DESCRIPTION
**Pull request by tustvold**
Previously IOx would report the non-null count of the first column as the row count for the chunk, this is incorrect.

### CODE REVIEW

File: server/src/db/catalog/chunk.rs [hunk 1]
```diff
@@ -686,7 +686,7 @@ impl CatalogChunk {
                 read_buffer,
                 ..
             } => {
-                if let Some(read_buffer_inner) = &read_buffer {
+                if let Some(rub_chunk) = read_buffer.take() {
```
Comment (tustvold) [inline server/src/db/catalog/chunk.rs:689]: Unrelated drive-by cleanup

File: tests/end_to_end_cases/persistence.rs [hunk 2]
```diff
@@ -80,6 +87,10 @@ async fn test_chunk_are_removed_from_memory_when_soft_limit_is_hit() {
         chunks.len(),
         chunks
     );
+
+    for chunk in &chunks {
+        assert_eq!(chunk.row_count, 1000)
```
Comment (tustvold) [inline tests/end_to_end_cases/persistence.rs:92]: Previously this would report 500

File: tests/end_to_end_cases/persistence.rs [hunk 3]
```diff
@@ -51,11 +52,17 @@ async fn test_chunk_are_removed_from_memory_when_soft_limit_is_hit() {
     // (as of time of writing, 10 chunks took up ~800K)
     let num_chunks = 10;
     for _ in 0..num_chunks {
-        let lp_lines: Vec<_> = (0..1_000)
-            .map(|i| format!("data,tag1=val{} x={} {}", i, i * 10, i))
-            .collect();
+        let lp_data = (0..500)
```
Comment (tustvold) [inline tests/end_to_end_cases/persistence.rs:55]: This formatting is a little weird, but is going to get replaced by #1776

**Review by e-dard — APPROVED** [review]

**Review by crepererum — APPROVED** [review]

**Review by alamb — COMMENTED** [review]
👍
